lib/deployment_runner.py

```diff
@@ -110,6 +123,186 @@ def rhv_storage(self, page):
+        page.set_ssh_password(self.osp.undercloud_pass)
+        page.click_detect_undercloud()
+        return page.click_next()
+
+    def osp_register_nodes(self, page):
+        '''
+        OSP Register Nodes Page Filler
+
+        This one is pretty complicated.   First we have to click register
+        nodes button.   This opens a modal, where one can fill in various ways
+        of selecting our nodes.   Presently, we only support doing autodetect.
+        Information to do autodetec is filled in and then Next is selected, but
+        this takes you to a new modal where you select which machines to
+        register.
+
+        TODO: The yaml and software supports registering nodes from multiple
+              Hosts, however this code only supports one Host presently.
+              At some point we should make it support multiple hosts.
+        '''
+
+        ##################################
+        # The Register Nodes Modal Frame #
+        ##################################
+        for node_registration_info in self.osp.overcloud_nodes:
+            page.click_register_nodes()
+            self._osp_determine_how_to_register_nodes(
+                page,
+                node_registration_info
+            )
+
+        # Wait for the spinners to be done:
+        # XXX: I thought sometimes the wait would kick off before
+        # the spinners started, so we have this arbitrary sleep.
+        # We might be able to remove this.
+        time.sleep(3)
+        page.wait_on_spinner(
+            timeout=300,
+            spin_class='spinner-xs'
+        )
+
+        return page.click_next()
+
+    def _osp_determine_how_to_register_nodes(self, page, node_registration_info):
+        '''
+        Handles filling out information for the:
+
+            Determine How to Register Nodes:
+
+        Modal frame that is part of the RHOSP:Register Nodes Page.
+        '''
+        host_ip = node_registration_info['host_ip']
+        driver_type = node_registration_info['driver_type']
+        host_username = node_registration_info['host_username']
+        host_password = node_registration_info['host_password']
+        mac_address = node_registration_info['mac_address']
+
+        # Click on "autodect or specify nodes" so we can fill the form out
+        # to manually specify the nodes.
+        # TODO: Need to add support for CSV file.
+        # TODO: Need to add support for autodetection.
+        page.click_autodetct_or_specify()
+
+        # Fill in all the information up to the autodetect slider.
+        page.set_ip_address(host_ip)
+        drivers_selector = page.driver
+        drivers_selector.select_by_value(driver_type)
+        page.set_username(host_username)
+        page.set_password(host_password)
+
+        # We will force autodetect off.
+        # TODO: Eventually we will need to update the config to make this
+        #      configurable so we can test autodetect.
+        if page.is_autodetect_enabled:
```

Comment:
This all messed up.   My is_autodetect_enabled is not working.   I'm going to look a little closer and see if I can figure this out today.   I should have left an XXX comment (the traditional redhat way of saying warning there's a hack here) so at least you knew why I was doing this.  If its taking too long to figure out, I'll create a Jira card for it and get it after this is submitted (it works right now, it's just wrong).